core/Pos/Util/Util.hs

```diff
@@ -260,6 +261,10 @@ type instance ChannelT (Ether.TaggedTrans tag t m) = ChannelT m
 maybeThrow :: (MonadThrow m, Exception e) => e -> Maybe a -> m a
 maybeThrow e = maybe (throwM e) pure
 
+-- | Fail or return result depending on what is stored in 'Either'.
+eitherToFail :: (MonadFail m, ToString s) => Either s a -> m a
```

Comment:
@int-index but `MonadError Text m` is used like `MonadFail` in this PR (`catch` isn't used).
so I don't understand advantages `MonadError Text` here